### PR TITLE
[WIP] Add support for Julia 

### PR DESCRIPTION
**Pull request by findmyway**
Hi guys,

I would like to add a Julia wrapper for VisualDL.

Previously I've written a [wrapper](https://github.com/Ju-jl/VisualDL.jl) for VisualDL by leveraging [PyCall.jl](https://github.com/JuliaPy/PyCall.jl). Obviously there's a little overhead.

Recently I have spent some time on how to use the C++ SDK of VisualDL by using [CxxWrap.jl](https://github.com/JuliaInterop/CxxWrap.jl). And it works fine.

This PR should give you a glimpse of the up-coming changes to support Julia. Before updating `vdljl.cc` to support more datatypes (histogram, text, voice...), I would like to hear your thoughts/suggestions.

Cheers!

### CODE REVIEW

File: visualdl/julia/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+find_package(JlCxx REQUIRED)
```
Comment (nepeplwu) [inline visualdl/julia/CMakeLists.txt:1]: Is it more appropriate to provide a cmake file(such as cmake/external/JlCxx.cmake) ?

File: visualdl/julia/vdljl.cc [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                 jlcxx::ArrayRef<cp::Image::value_t, 1> data) {
+                std::vector<cp::Image::shape_t> s(shape.begin(), shape.end());
+                std::vector<cp::Image::value_t> d(data.begin(), data.end());
+                return img.SetSample(index, s, d);
+              });
+
+  vdl.add_type<cp::Text>("Text")
+      .constructor<vs::Tablet>()
+      .method("set_caption", &cp::Text::SetCaption)
+      .method("AddRecord", &cp::Text::AddRecord);
```
Comment (nepeplwu) [inline visualdl/julia/vdljl.cc]: a typo? I think "add_record" is more appropriate

**Comment by nepeplwu** [comment]
I wonder if vdl.jl is a demo or will be a julia sdk, if it is a demo, move it to demo dir is more reasonable

File: visualdl/julia/vdljl.cc (hunk 2 above)
Comment (findmyway) [reply visualdl/julia/vdljl.cc]: fixed

File: visualdl/julia/CMakeLists.txt (hunk 1 above)
Comment (findmyway) [reply visualdl/julia/CMakeLists.txt:1]: To make this package simple enough. I prefer to specify the JlCxx_DIR in the building stage.

**Comment by findmyway** [comment]
@nepeplwu 
Thanks for your review!
Yes, it's just a quick demo. And I'll remove it first and provide a more detailed demo later.

The whole logic is that:

1. Add CxxWrap.jl as a dependency in VisualDL.jl
1. In the building state of VisualDL.jl, the source code of VisualDL will be downloaded and compiled to a .so file. (JlCxx_DIR specified here)

Once this PR get merged. I'll update the README file to describe how to use it in julia (which is super easy!). Also if you agree, I can transfer VisualDL.jl to this org for better maintainance.

**Comment by nepeplwu** [comment]
@findmyway , Thank you for your contribution to VisualDL, this PR looks good to me. I recommend updating the README doc(which is in docs/release) and describe the entire step about building a julia-enable package(such as how to install JlCxx)

**Review by nepeplwu — APPROVED** [review]

**Comment by findmyway** [comment]
Thanks

Sure, once https://github.com/JuliaWeb/MbedTLS.jl/issues/193 get fixed, I'll make another PR to document the whole pipeline and provide some interesting demos.

**Pull request merged**